test/unit/core.js

```diff
@@ -663,6 +663,9 @@ QUnit.test( "jQuery('html')", function( assert ) {
 
 	assert.ok( !jQuery( "<script/>" )[ 0 ].parentNode, "Create a script" );
 
+	jQuery( "body" ).append( "<script nonce='test' id='nonceCreatedTest' />" );
+	assert.equal( jQuery( "#nonceCreatedTest" ).attr( "nonce" ), "test", "Ensure the nonce attribute is preserved for CSP" );
```

Comment:
This doesn't test the new functionality, and in fact already passes: http://jsfiddle.net/gwk04yht/

You need to do something _within_ the invoked script.